services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/PortHistorySpec.groovy

```diff
@@ -150,13 +167,148 @@ class PortHistorySpec extends HealthCheckSpecification {
+            assert islUtils.getIslInfo(isl).get().state == IslChangeType.DISCOVERED
+        }
+    }
+
+    def "System shows antiflap statistic in the ANTI_FLAP_DEACTIVATED event when antiflap is deactivated\
+ before collecting ANTI_FLAP_PERIODIC_STATS"() {
+        assumeTrue("It can't be run when antiflap.cooldown + flap_duration > port.antiflap.stats.dumping.interval.seconds",
+                antiflapCooldown + 3 < antiflapDumpingInterval)
+        //port up/down procedure is done once in this test, so it can't take more than 3 seconds
+        assumeTrue("At least 10 seconds should be available for changing port status", antiflapCooldown >= 10)
+        // assume that portDown/portUp can take some time on hardware env(no more than 10 seconds)
+
+        given: "A direct link"
+        def isl = getTopology().islsForActiveSwitches.find { !it.aswitch }
+        assumeTrue("Unable to find ISL for this test", isl as boolean)
+        def timestampBefore = System.currentTimeMillis()
+
+        when: "Execute port DOWN on the src switch for activating antiflap"
+        northbound.portDown(isl.srcSwitch.dpId, isl.srcPort)
+        Wrappers.wait(WAIT_OFFSET) {
+            assert islUtils.getIslInfo(isl).get().state == IslChangeType.FAILED
+            Long timestampAfterDown = System.currentTimeMillis()
+            with(northboundV2.getPortHistory(isl.srcSwitch.dpId, isl.srcPort, timestampBefore, timestampAfterDown)) {
+                it.size() == 2 // PORT_DOWN, ANTI_FLAP_ACTIVATED
+            }
+        }
+
+        and: "Generate antiflap statistic"
+        northbound.portUp(isl.srcSwitch.dpId, isl.srcPort)
+        northbound.portDown(isl.srcSwitch.dpId, isl.srcPort)
+
+        then: "Antiflap statistic is available in port history inside the ANTI_FLAP_DEACTIVATED event"
+        Wrappers.wait(antiflapCooldown + WAIT_OFFSET) {
+            Long timestampAfterStat = System.currentTimeMillis()
+            with(northboundV2.getPortHistory(isl.srcSwitch.dpId, isl.srcPort, timestampBefore, timestampAfterStat)) {
+                it.findAll { it.event == PortHistoryEvent.ANTI_FLAP_PERIODIC_STATS.toString() }.empty
+                it.size() == 3 // PORT_DOWN, ANTI_FLAP_ACTIVATED, ANTI_FLAP_DEACTIVATED
+                def antiflapStat = it[-1]
+                checkPortHistory(antiflapStat, isl.srcSwitch.dpId, isl.srcPort,
+                        PortHistoryEvent.ANTI_FLAP_DEACTIVATED)
+                antiflapStat.downCount == 1
+                antiflapStat.upCount == 1
+            }
+        }
+
+        and: "Cleanup: revert system to original state"
+        northbound.portUp(isl.srcSwitch.dpId, isl.srcPort)
+        Wrappers.wait(WAIT_OFFSET + discoveryInterval + antiflapCooldown) {
+            assert islUtils.getIslInfo(isl).get().state == IslChangeType.DISCOVERED
+        }
+    }
+
+    @Tags(LOW_PRIORITY)
+    def "System doesn't return antiflap statistic in port history if no flapping occurs during cooldown"() {
```

Comment:
And why do you need a separate test for this if there is already a test without antiflap?

Reply:
you know better)
needed check was added to the first test